### PR TITLE
Fix the hash in the submitTx response

### PR DESCRIPTION
**Pull request by tok-kkk**

### CODE REVIEW

File: compat/v0/compat.go [hunk 1]
```diff
@@ -159,7 +163,7 @@ func TxFromV1Tx(t tx.Tx, hash B32, hasOut bool, bindings txengine.Bindings) (Tx,
 		Value: toAddr,
 	})
 
-	tokenAddrRaw, err := bindings.TokenAddressFromAsset(multichain.Ethereum, multichain.BTC)
```
Comment (TheAncientGoat) [inline compat/v0/compat.go:162]: Good catch on that one

File: compat/v0/compat.go [hunk 2]
```diff
@@ -68,8 +68,8 @@ func ShardsResponseFromState(state jsonrpc.ResponseQueryState) (ResponseQuerySha
 	return resp, nil
 }
 
-// TxFromV1Tx takes a V1 Tx and converts it to a V0 Tx
-func TxFromV1Tx(t tx.Tx, hash B32, hasOut bool, bindings txengine.Bindings) (Tx, error) {
+// TxFromV1Tx takes a V1 Tx and converts it to a V0 Tx, the given tx has to be a mint tx.
```
Comment (TheAncientGoat) [inline compat/v0/compat.go:71]: Hmm, let me double check this, might be problematic if we only support mint txes

**Review by TheAncientGoat — COMMENTED** [review]
Looks good, I'd just want to investigate if we query burn txes, because not being able to cast would break things.

Also might be good to have a test or two that demonstrates the changes

File: compat/v0/compat.go (hunk 2 above)
Comment (tok-kkk) [reply compat/v0/compat.go]: I believe this has been fixed in #119

**Pull request merged**